src/Text/Pandoc/Readers/Org.hs

```diff
@@ -407,11 +407,21 @@ orgBlock = try $ do
       _         -> withParsed (fmap $ divWithClass blkType)
 
 blockHeaderStart :: OrgParser (Int, String)
-blockHeaderStart = try $ (,) <$> indent <*> blockType
+blockHeaderStart = try $ (,) <$> spaceLength <*> blockType
  where
-  indent    = length      <$> many spaceChar
   blockType = map toLower <$> (stringAnyCase "#+begin_" *> orgArgWord)
 
+spaceLength :: OrgParser Int
```

Comment:
The "true" functional way would be to split this into two functions, e.g.
`indentation :: OrgParser Int` and `spaceLength :: Int String -> Int`.  The latter function would get the number of spaces per tab and the string as input, returning the converted number of whitspace.